### ISSUE TITLE
Dangling HREF on save

### ISSUE THREAD

**Issue by CamilleLetavernier** [eclipsesource/ecore-glsp#71]
With the new support for typing EAttributes (Issue #37), it is possible to get EMF "Dangling HREF" exceptions on save:

```
Caused by: org.eclipse.emf.ecore.xmi.DanglingHREFException: The object 'org.eclipse.emf.ecore.impl.EEnumImpl@103ed126 (name: NewEEnum7) (instanceClassName: null) (serializable: true)' is not contained in a resource.
	at org.eclipse.emf.ecore.xmi.impl.XMLHelperImpl.handleDanglingHREF(XMLHelperImpl.java:754)
	at org.eclipse.emf.ecore.xmi.impl.XMLHelperImpl.getURIFragment(XMLHelperImpl.java:725)
	at org.eclipse.emf.ecore.xmi.impl.XMLHelperImpl.getIDREF(XMLHelperImpl.java:747)
	at org.eclipse.emf.ecore.xmi.impl.XMLSaveImpl.saveIDRefSingle(XMLSaveImpl.java:1992)
```

The exception happens when an Element (In this case, an EEnum) is referenced by another element (An EAttribute), and the referenced element is deleted: the EAttribute references an element that no longer exists, which causes this exception on save.

**Comment by CamilleLetavernier** [comment]
To reproduce:

- Create an EAttribute
- Create a type (EDataType or EEnum)
- Type the EAttribute with this type
- Delete the type
- Try to save

**Comment by CamilleLetavernier** [comment]
We need two actions to solve the issue & make the save more robust in the future:

- Find & Unset cross-references when deleting an element
- Add an EMF Save option to avoid breaking the model on save when dangling Hrefs are found (See XMLResource.OPTION_PROCESS_DANGLING_HREF)

**Comment by CamilleLetavernier** [comment]
> Find & Unset cross-references when deleting an element

See org.eclipse.emf.ecore.util.ECrossReferenceAdapter.getInverseReferences(EObject)

It's not the most efficient method, but Ecore models are usually small enough that it shouldn't really matter

**Comment by tortmayr** [comment]
> Find & Unset cross-references when deleting an element

@CamilleLetavernier  If I'm not mistaken  [ EcoreUtil.delete](https://download.eclipse.org/modeling/emf/emf/javadoc/2.5.0/org/eclipse/emf/ecore/util/EcoreUtil.html) provides exactly that behavior. So this potentially can be resoved by using  `EcoreUtil.delete` instead of `EcoreUtil.remove` in the `EcoreDeleteOperationHandler`

**Comment by CamilleLetavernier** [comment]
:+1:

That would solve the first point :) I would still add an extra safety with the save option

**Comment by CamilleLetavernier** [comment]
Fixed with #137
